### PR TITLE
R - move findFileName from FuncForPC to CallerFrames

### PR DESCRIPTION
**Pull request by karngyan**
Closes [40](https://github.com/approvals/go-approval-tests/issues/40)

## Description

[Go runtime docs discourage traversing PC or FuncForPC on any returned PCs ](https://pkg.go.dev/runtime#Callers)as they can't account for inlining or return program counter adjustment. This leads to wrong file names for approved and received files.

This PR refactors the implementation of `findFileName`. I came across this issue while working on one of our approval tests at my company and have been struggling to reproduce it using simple examples. But it did solve the issue we faced. 

## The solution

Implement the method with CallerFrames instead.

### CODE REVIEW

File: approval_name.go [hunk 1]
```diff
@@ -48,28 +48,33 @@ func NewApprovalName(name string, fileName string) ApprovalName {
 func findFileName() (*string, error) {
 	pc := make([]uintptr, 100)
 	count := runtime.Callers(0, pc)
+	frames := runtime.CallersFrames(pc[:count])
 
-	i := 0
-	var lastFunc *runtime.Func
+	var lastFrame *runtime.Frame
+	var testFrame *runtime.Frame
```
[the remark this answers is not in the document]
Comment (karngyan) [reply approval_name.go]: Done.

**Comment by karngyan** [comment]
> Looks good but could you rebase ? I found that it breaks a test that was introduced recently because findFileName becomes case sensitive

Which test case are you talking about, and also what recent change?

**Comment by hjwk** [comment]
> Which test case are you talking about, and also what recent change?

TestVerifyJSONBytesWithRegexScrubber fails
`scrubber_test.go:63: Failed Approval: received does not match approved.`

Tested on linux (Pop_os) and go 1.18.3

**Comment by karngyan** [comment]
> `scrubber_test.go:63: Failed Approval: received does not match approved.`

Hey, I think it's rebased already. I don't see any test failures. Can you please confirm you're running the right branch?

Screenshot 2022-08-23 at 7 22 17 PM



Even ended up running the workflow which passed as well: https://github.com/karngyan/go-approval-tests/actions/runs/2911761732

**Review by hjwk — APPROVED** [review]

**Comment by aleclerc-cio** [comment]
@hjwk I know there hasn't been a commit to this project in a little while. Is there any chance of getting this PR merged?

**Comment by hjwk** [comment]
sorry, I don't have write access

**Pull request merged**